### PR TITLE
Conv, MaxPool, AveragePool

### PR DESCRIPTION
**Pull request by varunjain99**
Translation for Conv, MaxPool, and AveragePool
Note: count_include_pad = 0 has no tests so I don't support it for now
MaxPool was also changed yesterday to have optional outputs and storage order attribute which I don't support for now

@bddppq @yinghai @Maratyszcza

### CODE REVIEW

File: onnx_xla/translations/translate_average_pool.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  XlaOp sumOp = builder.ReduceWindowWithGeneralPadding(
+      valueToOp.at(n.inputs().at(0)),
+      builder.ConstantLiteral(Literal::Zero(dataType)), add(dataType),
+      helper.getWindowDimensions(), helper.getWindowStrides(),
+      helper.getInputPadding());
+
+  // Build average with implicit broadcasting (if count_include_pad != 0)
+  // TODO: Support for count_include_pad (No test cases right now)
+  auto kcount_include_pad = Symbol("count_include_pad");
+  if (n.hasAttribute(kcount_include_pad) && n.i(kcount_include_pad) == 0) {
```
Comment (bddppq) [inline onnx_xla/translations/translate_average_pool.cc]: the default value is also 0

Comment (varunjain99) [reply onnx_xla/translations/translate_average_pool.cc]: Are you sure? It's passing the test cases with the assumption that the default is 1.

Comment (bddppq) [reply onnx_xla/translations/translate_average_pool.cc]: https://github.com/onnx/onnx/blob/master/onnx/defs/nn/defs.cc#L265

File: tensorflow.patch [hunk 2]
```diff
@@ -139,8 +143,20 @@ index 0000000000..ff7b11cf1d
...
 +#endif  // TENSORFLOW_COMPILER_XLA_RPC_COMPUTATION_CLIENT_H_
+diff --git a/tensorflow/compiler/xla/rpc/grpc_service_main.cc b/tensorflow/compiler/xla/rpc/grpc_service_main.cc
+index c68c857..7110c70 100644
+--- a/tensorflow/compiler/xla/rpc/grpc_service_main.cc
++++ b/tensorflow/compiler/xla/rpc/grpc_service_main.cc
+@@ -48,6 +48,7 @@ int RealMain(int argc, char** argv) {
+ 
+   builder.AddListeningPort(server_address, ::grpc::InsecureServerCredentials());
+   builder.RegisterService(service.get());
++  builder.SetMaxReceiveMessageSize(INT_MAX);
```
Comment (bddppq) [inline tensorflow.patch]: Is this still needed?

**Comment by yinghai** [comment]
What's the issue with shufflenet?

**Comment by varunjain99** [comment]
XLA doesn't support grouped convolution. My implementation of separate convolution and concatenate turns out to be quite slow, especially when group number is high. This is magnified in shufflnet because there are convolutions with hundreds of groups. I think either we'll have to come up with more efficient translation of groups or XLA will need to support grouped convolution operator.

File: tensorflow.patch (hunk 2 above)
Comment (varunjain99) [reply tensorflow.patch]: Yes, otherwise server will complain the message is too large.

**Comment by Maratyszcza** [comment]
@varunjain99 You may expand grouped convolution into full convolution with block-sparse weights. Many NN accelerators can skip zero-weights, so it wouldn't be as bad as it sounds.

**Comment by varunjain99** [comment]
I'll give it a try! @Maratyszcza

**Pull request merged**